src/features/game/parser/QuizParser.ts

```diff
@@ -0,0 +1,132 @@
+  }
+
+  /**
+   * This function parses the quiz's questions,
+   * converts each into a Question object,
+   * and store in an array.
+   *
+   * @param map The map of raw questions that map from index to question body
+   */
+  private static parseQuizQuestions(map: Map<string, string[]>): Question[] {
+    const questions: Question[] = new Array(map.size);
+    map.forEach((value: string[], key: string) => {
+      questions[parseInt(key)] = this.createQuestion(value);
+    });
+    return questions;
+  }
+
+  /**
+   * This function parses a question and
+   * create a Question object.
+   *
+   * @param questionText The text of a question body
+   * containing question text, correct answer, and options
+   */
+  private static createQuestion(questionText: string[]): Question {
+    if (questionText.length < 2) {
+      throw new Error('Parsing error: Quiz missing question or answer');
+    }
+    const question: Question = {
+      question: questionText[0],
+      speaker: SpeakerParser.parse('@narrator'),
+      answer: this.getQuizAnswer(questionText[1]),
+      options: this.parseOptions(questionText.slice(2))
+    };
+    return question;
+  }
+
+  /**
+   * This function parses a answer string and
+   * converts it into Number.
+   *
+   * @param answer The string containing the correct answer of a question
+   */
+  private static getQuizAnswer(answer: string): Number {
+    const ans = answer.split(':');
+    if (ans.length < 2 || Number.isNaN(parseInt(ans[1]))) {
+      throw new Error('Parsing error: Invalid answer for Quiz');
+    }
+    return parseInt(ans[1]);
+  }
+
+  /**
+   * This function parses options of a question and
+   * store them in an array.
+   *
+   * @param optionsText An Array of string containing all options' content,
+   * including option text and reactions
+   */
+  private static parseOptions(optionsText: string[]): Option[] {
+    const optionsParagraph = StringUtils.splitToParagraph(optionsText);
+    const options: Option[] = Array(optionsParagraph.length);
+    optionsParagraph.forEach(([header, content]: [string, string[]], index) => {
+      options[index] = this.createOption(content);
+    });
+    return options;
+  }
+
+  /**
+   * This function creates an Option object.
+   *
+   * @param content An Array of string containing an option's content,
+   * including option text and reaction
+   * @param [noReaction=false] Indicates whether this option provides a reaction
+   */
+  private static createOption(content: string[], noReaction: boolean = false): Option {
+    if (!content) {
+      throw new Error('Parsing error: Quiz option not provided');
+    }
+    if (content.length <= 1) {
+      noReaction = true;
+    }
+    const option: Option = {
+      text: content[0],
+      reaction: noReaction ? undefined : DialogueParser.parseQuizReaction(content.slice(1))
+    };
+    return option;
+  }
+}
+
+const isInteger = (line: string) => new RegExp(/^[0-9]+$/).test(line);
```

Comment:
We could also use [`isInteger`](https://www.geeksforgeeks.org/lodash-_-isinteger-method/) from `lodash` - thoughts? 


(IMO not too big of an issue but we can at least avoid unnecessary code)

Reply:
The current `isInteger` detects strings containing only digits, so is not the same as `isInteger` from `lodash`.
The function is copied from other parser files. Will it be better to move them to some Utils file and import from there?